lib/protocol/counting.go

```diff
@@ -10,8 +10,8 @@ import (
 
 type countingReader struct {
 	io.Reader
-	tot  int64 // bytes
-	last int64 // unix nanos
+	tot  int64 // bytes (atomic, must remain 64-bit aligned)
+	last int64 // unix nanos (atomic, must remain 64-bit aligned)
 }
```

Comment:
I wonder if we need to move the interface below

Reply:
As stated in the commit description, this is not required.

The size of an interface is 2 machine-word long, for all platforms, on a 32-bit architecture such as arm sizeof(interface) = 64-bit, implying that `tot` and `last` offset are, respectively, 64 and 128 bits from the top of the struct.

All those values are guaranteed with the standard Go compiler: [go/types/sizes.go](https://golang.org/src/go/types/sizes.go?s=4090:4130)